.github/workflows/reuse.yaml

```diff
@@ -0,0 +1,134 @@
+
+on:
+  workflow_call:
+    inputs:
+      publish_command:
+        required: true
+        type: string
+      digest_command: 
+        required: true
+        type: string
+      image_name: 
+        required: true
+        type: string
+      tag: 
+        required: true
+        type: string
+    secrets:
+      registry_username:
+        required: true
+      registry_password:
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    steps:
+      - name: Checkout release
+        if: ${{ inputs.tag == 'release'}}
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        with:
+          fetch-depth: 0
+
+      - name: Checkout image
+        if: ${{ inputs.tag == 'image'}}
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Unshallow
+        if: ${{ inputs.tag == 'image'}}
+        run: git fetch --prune --unshallow --tags
+
+      - name: Set up Go
+        uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 # v2.1.5
+        with:
+          go-version: 1.16
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@116dc6872c0a067bcb78758f18955414cdbf918f # v1.4.1
+        with:
+          cosign-release: 'v1.4.1'
+
+      - name: Cache Go modules
+        uses: actions/cache@d9747005de0f7240e5d35a68dca96b3f41b8b340 # v1.2.0
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Refresh Go report card
+        if: ${{ inputs.tag == 'release'}}
+        uses: creekorful/goreportcard-action@1f35ced8cdac2cba28c9a2f2288a16aacfd507f9 # v1.0
+
+      - name: Log into ghcr.io
+        uses: docker/login-action@master
+        with:
+          registry: ghcr.io
+          username: ${{secrets.registry_username}}
+          password: ${{secrets.registry_password}}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4ab11c41e45d028884a99163086648e898eed25 # v1.6.0
+        id: buildx
+        with:
+          install: true
+
+      - name: Set Version
+        if: ${{ inputs.tag == 'release'}}
+        run: |
+          echo "KYVERNO_VERSION=$(git tag --list --sort=-version:refname "v[0-9]*" | head -n 1)" >> $GITHUB_ENV
+
+      - name: Generate SBOM JSON
+        if: ${{ inputs.image_name == 'kyverno' && inputs.tag == 'release'}}
+        uses: CycloneDX/gh-gomod-generate-sbom@v1
+        with:
+          version: v1
+          args: mod -licenses -json -output kyverno-${{ env.KYVERNO_VERSION }}-bom.cdx.json
+          
+      - name: Upload SBOM JSON
+        if: ${{ inputs.image_name == 'kyverno' && inputs.tag == 'release'}}
+        uses: actions/upload-artifact@v2
+        with:
+          name: kyverno-bom-cdx
+          path: kyverno-v*-bom.cdx.json
```

Comment:
Can we remove the conditional run on `inputs.image_name == 'kyverno'` and upload it for all 3 images that are published out of this repository i.e. kyverno, kyvernopre and kyverno-cli?

This would also mean renaming the output SBOM based on the input image.

It is equally useful to have the SBOMs for the other kyverno images.

Reply:
SBOM would be the same for all three images, right? Then how is it useful to upload 3 similar SBOMs?

Reply:
Nope - it won't be the same for all 3 images. The cyclonedx action has an app mode, which produces an SBOM only out of the packages that were actually used in the output binary rather than the ones declared in go.mod.

Reply:
See https://github.com/CycloneDX/cyclonedx-gomod#app

Reply:
> Distributors of applications will typically use "app" and provide the resulting SBOMs
alongside their application's binaries. This enables users to only consume SBOMs for
artifacts that they actually use. For example, a Go module may include "server" and
"client" applications, of which only the "client" is distributed to users. 
Additionally, modules included in "client" may differ, depending on which platform 
it was compiled for

Reply:
Got it 👍 Thanks for the clarification, making the required changes right away